src/shared.js

```diff
@@ -801,9 +801,10 @@ export const getBoxesByLocation = async (location) => {
     return res;
 }
 
-export const searchSpecimen = async (masterSpecimenId) => {
+export const searchSpecimen = async (masterSpecimenId, allSitesFlag) => {
     const idToken = await getIdToken();
-    const response = await fetch(`${api}api=searchSpecimen&masterSpecimenId=${masterSpecimenId}`, {
+    const specimenQuery = allSitesFlag === true ? `&masterSpecimenId=${masterSpecimenId}&allSitesFlag=${allSitesFlag}`: `&masterSpecimenId=${masterSpecimenId}`
```

Comment:
Comment only...Just another way to build the string (factor out the common parts):
```javascript
const specimenQuery = `&masterSpecimenId=${masterSpecimenId}` + (allSitesFlag ? `&allSitesFlag=${allSitesFlag}` : '');
```